internal/web/routes/routes.go

```diff
@@ -0,0 +1,18 @@
+package routes
+
+import (
+	"context"
+
+	"github.com/google/go-github/v68/github"
+	"github.com/labstack/echo/v4"
```

Comment:
There is a logical error with the route parameters. In the `/commits/repo/:repo/pulls/:pull_number/owner/:owner/` route and others, ensure parameters are correctly validated. If the parameters are invalid, it could lead to unintended behavior when processing requests.